spec/intl.html

```diff
@@ -2365,8 +2365,14 @@ <h1>PrepareTemporalFields ( _fields_, _fieldNames_, _requiredFields_ )</h1>
                 1. Let _Conversion_ represent the abstract operation named by the Conversion value of the same row.
                 1. Set _value_ to ? _Conversion_(_value_).
             1. Perform ! CreateDataPropertyOrThrow(_result_, _property_, _value_).
-          1. Let _era_ be ? Get(_result_, *"era"*).
-          1. Let _eraYear_ be ? Get(_result_, *"eraYear"*).
+          1. If ! HasOwnProperty(_result_, *"era"*) is *true*, then
+            1. Let _era_ be ! Get(_result_, *"era"*).
+          1. Else,
+            1. Let _era_ be *undefined*.
+          1. If ! HasOwnProperty(_result_, *"eraYear"*) is *true*, then
+            1. Let _eraYear_ be ! Get(_result_, *"eraYear"*).
+          1. Else,
+            1. Let _eraYear_ be *undefined*.
```

Comment:
Suggestion similar to [#2255](https://github.com/tc39/proposal-temporal/pull/2255/files#diff-e45ec7b978e8c3ddbd13852fa4387ba083d55f68a70822a73644d11fd6837007R762-R766):
```suggestion
          1. Let _era_ be OrdinaryGetOwnProperty(_result_, *"era"*).
          1. If _era_ is not *undefined*, then
            1. Assert: _era_ is a data Property Descriptor.
            1. Set _era_ to _era_.[[Value]].
          1. Let _eraYear_ be OrdinaryGetOwnProperty(_result_, *"eraYear"*).
          1. If _eraYear_ is not *undefined*, then
            1. Assert: _eraYear_ is a data Property Descriptor.
            1. Set _eraYear_ to _eraYear_.[[Value]].
```
(but if not accepted, I'll just update that PR to match this one)

Reply:
I wasn't sure how others feel about using `OrdinaryGetOwnProperty`, because sometimes it is considered too low-level. That's why I used the more general operations from [Abstract Operations](https://tc39.es/ecma262/#sec-abstract-operations).

There's another alternative which avoids `HasOwnProperty` / `OrdinaryGetOwnProperty` altogether. We could simply check if the names are in `fieldNames`: 

> 4. If fieldNames contains "era", then
>     a. Let era be ! Get(result, "era").
> 5. Else,
>     a. Let era be undefined.
> 6. If fieldNames contains "eraYear", then
>     a. Let eraYear be ! Get(result, "eraYear").
> 7. Else,
>     a. Let eraYear be undefined.